File: LeetCode/Study Plan/Practice Problems/Easy/Base 7/Cpp/base_seven.cc
#include <iostream>

std::string base_seven(int n) {
    bool neg = false;
    if (n<0) {
        n *= -1;
        neg = true;
    }
    if (n == 0 ) {
        return "0";
    }
    std::string s = "";
    while (n) {
        s = std::to_string(n%7) + s;
        n = n/7;
    }
    return neg ? "-" + s : s;
}

int main () {
    int n = 100;
    std::cout<<base_seven(n)<<"\n";
}

File: LeetCode/Study Plan/Practice Problems/Easy/Base 7/Go/base_seven.go
package main

import (
	"fmt"
	"strconv"
)

func base_seven(n int) string {
	str := ""
	neg := false
	if n == 0 {
		return "0"
	}
	if n < 0 {
		neg = true
		n = n * -1
	}
	for n > 0 {
		str = strconv.Itoa(n%7) + str
		n = n / 7
	}
	if neg {
		return "-" + str
	}
	return str
}

func main() {
	fmt.Println(base_seven(100))
}
